crypto/srp/srp_vfy.c

```diff
@@ -201,16 +201,20 @@ static SRP_user_pwd *SRP_user_pwd_new(void)
     return ret;
 }
 
-static void SRP_user_pwd_set_gN(SRP_user_pwd *vinfo, const BIGNUM *g,
-                                const BIGNUM *N)
+void SRP_user_pwd_set_gN(SRP_user_pwd *vinfo, const BIGNUM *g,
+                         const BIGNUM *N)
 {
     vinfo->N = N;
     vinfo->g = g;
 }
 
-static int SRP_user_pwd_set_ids(SRP_user_pwd *vinfo, const char *id,
-                                const char *info)
+int SRP_user_pwd_set1_ids(SRP_user_pwd *vinfo, const char *id,
+                         const char *info)
 {
+    if (vinfo->id != NULL)
+        OPENSSL_free(vinfo->id);
+    if (vinfo->info != NULL)
+        OPENSSL_free(vinfo->info);
```

Comment:
Drop the "if". Just call `OPENSSL_free` unconditionally. `OPENSSL_free` handles a NULL input.